### DISCUSSION TITLE
Uncaught (in promise) Component not found error after v3 upgrade

### DISCUSSION THREAD

**Discussion by jtomlinson** [livewire/livewire#9035] [1. Help]
I recently updated an application from Livewire v2 to v3 and have been seeing numerous `Uncaught (in promise) Component not found:` and `Component not found:` errors. I've done what the docs and others have said about adding `wire:key` and `:key` where needed to no avail.

I think I've narrowed it down to pages that make use of `wire:poll` or a page that sometimes has a slower response. Like in the following example, the page renders just fine and I've added a `sleep()` call to simulate a slow loading time. If a user clicks the div with `wire:click` the request is made to the server and if the user clicks on the same div or another div while the server is still processing, a `Uncaught (in promise) Component not found:` will be generated.

Is there something I'm missing that needs to be changed in the blade file or component itself? In my testing I have found that if I remove either the nested component (livewire:route-test) or the div that contains `x-cloak` the error does not occur. Is there something conflicting between these two items?

Any help would be greatly appreciated, as if I can't get this solved I will be forced to migrate back to Livewire v2.

A sample repo can be found here https://github.com/jtomlinson/livewire-test

### Routes.php
```php
class Routes extends Component
{
    public int $selectedRouteId = 0;

    public function render()
    {
        sleep(1);

        $routes = Route::paginate(10);

        return view('livewire.routes')
            ->with('routes', $routes)
            ->extends('blank');
    }

    public function selectRoute(int $routeId): void
    {
        $this->selectedRouteId = $routeId;
    }
}
```

### routes.blade.php
```php
<div x-data="routes">
    <flux:icon.home wire:loading.remove/>
    <flux:icon.loading wire:loading/>
    <div class="grid grid-cols-2 w-1/2">
        <div class="grid grid-cols-1 gap-y-1">
            @foreach ($routes as $route)
                <div class="flex w-full" wire:key="route-list-route-{{ $route->id }}">
                    <div data-id="{{ $route->id }}"
                         wire:click="selectRoute({{ $route->id }})"
                         wire:key="route-{{ $route->id }}"
                         class="w-full bg-blue-700 text-white text-xs items-center justify-between border border-gray-400 p-2 rounded cursor-pointer hover:bg-blue-600">
                        {{ $route->id }}
                    </div>
                </div>
            @endforeach
        </div>

        <div class="grid grid-cols-1 gap-y-2">
            @if ($this->selectedRouteId !== 0)
                <livewire:route-test :key="$this->selectedRouteId"/>
            @endif
        </div>
    </div>

    <div x-cloak :class="{'visible': moveToRouteModal.isVisible, 'invisible': !moveToRouteModal.isVisible }" wire:key="modal">
        <div x-show="moveToRouteModal.isVisible" wire:key="moveToRouteModal"></div>
    </div>
</div>

@script
<script type="text/javascript">
    Alpine.data('routes', () => ({
        moveToRouteModal: {
            isVisible: false,
            hide() {
                this.isVisible = false;
            },
            show() {
                this.isVisible = true;
            },
        }
    }))
</script>
@endscript

```

### RouteTest.php
```php
class RouteTest extends Component
{
    public function render()
    {
        return view('livewire.route-test');
    }
}
```

### route-test.blade.php
```php
<div>
    data
</div>
```

**Comment by jonathanbriehl** [comment]
Did you solve this? I'm having the same issue.

**Reply by ralphjsmit** [reply]
Having the same issue as well.

**Reply by joshhanley** [reply]
We've just done a bunch of work to try and fix the scenarios which trigger this error message. The updates will be in the next release.
